dotnet/samples/GettingStartedWithProcesses/Step01_Processes.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace GettingStartedWithProcesses;
+
+/// <summary>
+/// Demonstrate creation of <see cref="KernelProcess"/> and
+/// eliciting its response to three explicit user messages.
+/// </summary>
+public class Step01_Processes(ITestOutputHelper output) : BaseTest(output)
+{
+    [Fact]
+    public async Task UseSimpleProcessAsync()
+    {
+        // Create a kernel with a chat completion service
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddOpenAIChatCompletion(
+                modelId: TestConfiguration.OpenAI.ChatModelId,
+                apiKey: TestConfiguration.OpenAI.ApiKey)
+            .Build();
+
+        // Create a process that will interact with the chat completion service
+        ProcessBuilder process = new("ChatBot");
+        var introStep = process.AddStepFromType<IntroStep>();
+        var userInputStep = process.AddStepFromType<UserInputStep>();
+        var responseStep = process.AddStepFromType<ChatBotResponseStep>();
+
+        // Define the behavior when the process receives an external event
+        process.OnExternalEvent("StartProcess")
+            .SendEventTo(new ProcessFunctionTargetBuilder(introStep, "PrintIntroMessage"));
+
+        // When the intro is complete, notify the userInput step
+        introStep
+            .OnFunctionResult("PrintIntroMessage")
+            .SendEventTo(new ProcessFunctionTargetBuilder(userInputStep, "GetUserInput"));
+
+        // When the userInput step emits an exit event, send it to the end steprt
+        userInputStep
+            .OnFunctionResult("GetUserInput")
+            .StopProcess();
+
+        // When the userInput step emits a user input event, send it to the assistantResponse step
+        userInputStep
+            .OnEvent(ChatBotEvents.UserInputReceived)
+            .SendEventTo(new ProcessFunctionTargetBuilder(responseStep, "GetChatResponse", "userMessage"));
+
+        // When the assistantResponse step emits a response, send it to the userInput step
+        responseStep
+            .OnEvent(ChatBotEvents.AssistantResponseGenerated)
+            .SendEventTo(new ProcessFunctionTargetBuilder(userInputStep, "GetUserInput"));
+
+        // Build the process to get a handle that can be started
+        KernelProcess kernelProcess = process.Build();
+
+        // Start the process with an initial external event
+        var runningProcess = await kernelProcess.StartAsync(kernel, new KernelProcessEvent() { Id = "StartProcess", Data = null });
+    }
+
+    public class IntroStep : KernelProcessStep
+    {
+        [KernelFunction()]
+        public void PrintIntroMessage()
+        {
+        }
+    }
+
+    public class UserInputStep : KernelProcessStep<UserInputState>
+    {
+        private UserInputState? _state;
```

Comment:
Should `KernelProcessStepState` be stored in `KernelProcessStep`

Consider this pattern

```csharp
class KernelProcessStep
{
  KernelProcessStepState State { get; private set; } 
  
  // Default implementation store the state
  ValueTask ActivateAsync(KernelProcessStepState state)
  {
    this.State = state;
    await this.HandleActivatedAsync(state);
    return ValueTask.CompletedTask;
  }

  // Subclasses an optionally override this method to initialise the state.
  ValueTask HandleActivatedAsync(KernelProcessStepState state)
  {
  }
}
```